contracts/liquidator/Liquidator.sol

```diff
@@ -61,41 +72,41 @@ contract Liquidator is IUniswapV3FlashCallback, PeripheryImmutableState, Periphe
      * @param _comet The Compound V3 Comet instance address
      * @param _factory The Uniswap V3 pools factory instance address
      * @param _WETH9 The WETH address
-     * @param _assets The suported collateral assets
-     * @param _poolFees The Uniswap V3 pool fee used to locate the pool address
-     * @param _lowLiquidity The array that specifies if asset has low liquidity asset-USDC(base token) pool
+     * @param _poolConfigs The configurations of Uniswap asset pools
      **/
     constructor(
         ISwapRouter _swapRouter,
         CometInterface _comet,
         address _factory,
         address _WETH9,
-        address[] memory _assets,
-        uint24[] memory _poolFees,
-        bool[] memory _lowLiquidity
+        uint256 _liquidationThreshold,
+        UniswapPoolConfig[] memory _poolConfigs
     ) PeripheryImmutableState(_factory, _WETH9) {
-        require(_assets.length == _poolFees.length, "Wrong input");
-        require(_assets.length == _lowLiquidity.length, "Wrong input");
-
         swapRouter = _swapRouter;
         comet = _comet;
         weth = _WETH9;
+        liquidationThreshold = _liquidationThreshold;
```

Comment:
what's the advantage of setting a minimum? (as opposed to just letting the transaction proceed as long as its profitable)

Reply:
running complex logic for basically no gains, saving gas this way

Reply:
personally, i'd let the caller decide whether it's worth it to them to call. but i don't feel strongly about it

Reply:
but they cannot decide in our contract. There is some residue left in the protocol, and there will attempt to buy it. I agree, it has its pros and cons.